File: analytics/src/main/java/io/appmetrica/analytics/impl/utils/validation/NonEmptyStringValidator.java
package io.appmetrica.analytics.impl.utils.validation;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import io.appmetrica.analytics.coreutils.internal.StringUtils;

public class NonEmptyStringValidator implements Validator<String> {

    private final String mObjectDescription;

    public NonEmptyStringValidator(@NonNull String objectDescription) {
        mObjectDescription = objectDescription;
    }

    @Override
    public ValidationResult validate(@Nullable String data) {
        if (StringUtils.isNullOrEmpty(data)) {
            return ValidationResult.failed(this, mObjectDescription + " is empty.");
        } else {
            return ValidationResult.successful(this);
        }
    }
}

File: core-utils/src/main/java/io/appmetrica/analytics/coreutils/internal/validation/NonEmptyStringValidator.kt
package io.appmetrica.analytics.coreutils.internal.validation

class NonEmptyStringValidator(
    private val objectDescription: String
) : Validator<String?> {

    override fun validate(data: String?): ValidationResult =
        if (data.isNullOrEmpty()) {
            ValidationResult.failed(this, "$objectDescription is empty.")
        } else {
            ValidationResult.successful(this)
        }
}
